### PR TITLE
scaffolder: Handle undefined step input

### PR DESCRIPTION
**Pull request by jhaals**
## Hey, I just made a Pull Request!

- Fixes task failures caused by undefined step input
- Add TaskWorker tests

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4653-ehkwtkh6vs8xpqllpqvuzpu76agsugep.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6034c90465f8176978f76e49

File: plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      output: {
+        result: '{{ steps.test.output.testOutput }}',
+      },
+    });
+    const task = await broker.claim();
+    await taskWorker.runOneTask(task);
+    const { events } = await storage.listEvents({ taskId });
+    const event = events.find(e => e.type === 'completion');
+
+    if (!event) {
```
Comment (freben) [inline plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts:77]: So, `expect(event).toBeDefined()` then? :)

File: plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    });
+
+    const task = await broker.claim();
+    await taskWorker.runOneTask(task);
+
+    const { events } = await storage.listEvents({ taskId });
+    const event = events.find(e => e.type === 'completion');
+    if (!event) {
+      throw new Error('Expected event');
+    }
```
Comment (freben) [inline plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts:109]: Same

**Review by freben — APPROVED** [review]

File: plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts (hunk 1 above)
Comment (jhaals) [reply plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts:77]: That's a great idea however typescript remains unhappy saying might be undefined. 
Would you still opt for toBeDefined and then do `event!.body?` on the event?

Comment (freben) [reply plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts:77]: or `event?.body?.error`, either way

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4653-ehkwtkh6vs8xpqllpqvuzpu76agsugep.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6034c90465f8176978f76e49

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4653-ehkwtkh6vs8xpqllpqvuzpu76agsugep.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6034c90465f8176978f76e49

**Review by benjdlambert — APPROVED** [review]

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4653-ehkwtkh6vs8xpqllpqvuzpu76agsugep.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6034c90465f8176978f76e49

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4653-ehkwtkh6vs8xpqllpqvuzpu76agsugep.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6034c90465f8176978f76e49

**Pull request merged**